devito/petsc/logging.py

```diff
@@ -0,0 +1,133 @@
+from collections import namedtuple, OrderedDict
+from devito.petsc.iet.logging import PetscInfo
+
+
+class PetscEntry:
+    def __init__(self, **kwargs):
+        self._property_names = {k.lower(): k for k in kwargs.keys()}
+        self._properties = kwargs
+
+    def __getattr__(self, name):
+        """
+        Allow case insensitive attribute access.
+        """
+        original = self._property_names.get(name.lower())
+        if original:
+            return self._properties[original]
+        return object.__getattr__(self, name)
```

Comment:
This is still not quite right and I am still discouraging you from continuing to do this. Specifically here `object` is implicitly the build in python `object`, and should probably be `super`